### ISSUE TITLE
cl : command line warning D9025: overriding '/W4' with '/w'

### ISSUE THREAD

**Issue by Andreas-Schniertshauer** [google/benchmark#1209]
When compiling benchmark-1.5.6.zip I get lot's of warnings:
`cl : command line warning D9025: overriding '/W4' with '/w'
`
 in various .vcxproj files e.g. GoogleBenchmark\test\output_test_helper.vcxproj

**System**
  - OS:  Windows 10 21H1
  - Compiler and version: Visual Studio Professional 2019 16.11.0

This is because in various project properties the flag -w is set in additional options:
`C/C++/Command Line/Additional Options/%(AdditionalOptions) -w`
This flag comes from cmake\GoogleTest.cmake line 32+33:
`
add_compile_options(-w)
`

It seems you want to suppress the warnings:
`
check.h(42): warning C4722: 'benchmark::internal::CheckHandler::~CheckHandler': destructor never returns, potential memory leak
benchmark_register.h(26,1): warning C4244: '*=': conversion from 'int' to 'T', possible loss of data
gmock-matchers.h(366,12): warning C4244: 'argument': conversion from 'const M' to 'const Element', possible loss of data
`

This could be done at the location where the problem occurs with:
`
#pragma warning(push)
#pragma warning(disable:4244)
  ... code ...
#pragma warning(pop)
`

or for the whole project in additional options with:
`
/wd4244 /wd4722
`

I choose the second option and replaced in GoogleTest.cmake the line
`
add_compile_options(-w)
`

with
`
if (MSVC)
  add_compile_options(/wd4244 /wd4722)
endif()
`

Maybe you want to add the case for Linux too, but I don't know if this problem exists there too.
For me this code helped and I got no 4244 and 4722 warnings any more and the code compiled successfully.

**Comment by dominichamon** [comment]
ah yes, thanks. i would love to see a PR that adds the MSVC version to the existing linux `-w` flag.

**Comment by dominichamon** [comment]
can you take a look at #1320 please @Andreas-Schniertshauer, and confirm it's doing what you expect?

**Comment by Andreas-Schniertshauer** [comment]
@dominichamon Yes, I can confirm this works with benchmark 1.6.1 as expected, no more 4244 + 4722 warnings.
